scripts/tasks/publishPrepareTask.ts

```diff
@@ -8,11 +8,23 @@ export function publishPrepareTask() {
   const allInfo = getAllPackageInfo()
 
   for (const info of Object.values(allInfo)) {
-    if (info.packageJson.main && info.packageJson.main.startsWith('src/index')) {
+    if (info.packageJson.main && info.packageJson.main.startsWith('src/')) {
       const newPackageJson = {
         ...info.packageJson,
-        main: info.packageJson.main.replace('src/', 'lib/').replace(/\.tsx?/, '.js'),
       }
+
+      if (fs.existsSync(path.join(info.packagePath, 'just.config.ts'))) {
+        newPackageJson.main = info.packageJson.main.replace('src/', 'lib/').replace(/\.tsx?/, '.js')
+      } else {
+        const entry = info.packageJson.main.replace('src/', 'dist/es/').replace(/\.tsx?/, '.js')
+        Object.assign(newPackageJson, {
+          main: entry.replace('/es/', '/commonjs/'),
+          module: entry,
+          'jsnext:main': entry,
+          types: entry.replace('.js', '.d.ts'),
+        })
```

Comment:
Are you sure that this work for non-just packages?

Reply:
Yes, I tried running it locally and it worked as expected.